[Core][Parameters] Trying to remove the python iterator for `Parameters`

RiccardoRossi opened the pull request:
**📝 Description**

I want to see how many errors we get, in order to try addressing #10649

**🆕 Changelog**

- [Trying to remove the python iterator for parameters](https://github.com/KratosMultiphysics/Kratos/pull/10823/commits/4780f91461d9512b5dc9ae223d192e2055eb8969)

RiccardoRossi committed:
trting to remove the python iterator for parameters

i want to see how many errors we get, in order to try addressing #10649

RiccardoRossi committed:
removing a test of the iterators 

this is justified as we want to remove this feature

RiccardoRossi committed:
use .values() instead of iterators

RiccardoRossi committed:
using values() instead of iterators

RiccardoRossi committed:
using values()

RiccardoRossi committed:
use .values()

RiccardoRossi committed:
using .values()

RiccardoRossi committed:
changes in CompressiblePotentialFlowApplication and kratos

RiccardoRossi committed:
various corrections

RiccardoRossi committed:
using .values() instead of iterator

philbucher reviewed (changes requested):
> the problem is that you cannot deprecate this since it is in the python interface...

[it is](https://github.com/KratosMultiphysics/Kratos/pull/10835) ;)

RiccardoRossi committed:
using .values()

roigcarlo committed:
Update FluidDynamicsAnalysisMC.py

RiccardoRossi committed:
back to the version in master

this activates again the iterator, so that we can deprecate them as in #10835 

all the rest should go in (so that in essence we are not using the iterators anywhere in the master)

@philbucher with this i think you can accept it

philbucher commented on kratos/tests/test_kratos_parameters.py:
```diff
@@ -490,10 +490,10 @@ def test_iterators(self):
-        for iterator in kp:
-            nitems = nitems + 1
-        self.assertEqual(nitems, 5)
+        #nitems = 0
```
And please remove this

RiccardoRossi committed:
removing

RiccardoRossi committed:
removing

RiccardoRossi committed:
removing commented code

RiccardoRossi reviewed (commented):
@philbucher done!

philbucher commented on applications/SwimmingDEMApplication/python_scripts/parameters_tools.py:
```diff
@@ -54,7 +54,7 @@ def Condition(p, k):
     elif CanBeIterated(parameters): # there could be lists
 
-        for i, _ in enumerate(parameters):
+        for i, _ in enumerate(parameters.values()):
```
That one could be redone better I believe 🤔
FYI @sunethwarna

philbucher reviewed (approved):
Great!
